src/shader.cpp

```diff
@@ -751,6 +751,10 @@ ShaderInfo generate_shader(std::string name, u8 material_type, u8 drawtype,
 	if (g_settings->getBool("tone_mapping"))
 		shaders_header += "#define ENABLE_TONE_MAPPING\n";
 
+	shaders_header += "#define FOG_START ";
+	shaders_header += ftos(g_settings->getFloat("fog_start"));
```

Comment:
Does this need to be clamped between 2 values to avoid divide by zero? I see you've used rangelim in the engine but not limited the value in shaders?
Although advanced settings menu limits the value, when added in minetest.conf it can be set to anything.